### PR TITLE
#1000 Multiple Author Attribute

### LINKED ISSUE #1000: Support multiple Author attributes per test
Support multiple Author attributes per test

https://github.com/nunit/nunit/wiki/Author-Attribute

Note: you can currently only have one Author attribute per fixture or test.

### PR DESCRIPTION
**Pull request by JustinRChou**
resolves #1000 

Update to Author Attribute to allow for multiple use.

### CODE REVIEW

**Comment by JustinRChou** [comment]
I wasn't sure how should I go about testing the XML output of the authors.

And should TestCaseSource have the author property?

File: src/NUnitFramework/tests/Attributes/AuthorTests.cs [hunk 1]
```diff
@@ -95,5 +95,33 @@ public void AuthorOnTestCase()
...
+        [Test(Author = "Rob Prouse <rob@prouse.org>"),Author("Charlie Poole", "charlie@poole.org")]
+        [Author("NUnit")]
+        public void TestMethodMultipleAuthors()
+        {
+            Test test = TestBuilder.MakeTestFromMethod(FixtureType, "TestMethodMultipleAuthors");
+            var authors = test.Properties[PropertyNames.Author];
+            Assert.That(authors, Has.Exactly(3).Items);
+            Assert.AreEqual(authors[0], "Rob Prouse <rob@prouse.org>");
+            Assert.AreEqual(authors[1], "Charlie Poole <charlie@poole.org>");
+            Assert.AreEqual(authors[2], "NUnit");
```
Comment (CharliePoole) [inline src/NUnitFramework/tests/Attributes/AuthorTests.cs]: This is OK but I'd normally just do...

```C#
Assert.That(Test.Properties[PropertyNames.Author], Is.EqualTo(
    new [] { "Rob Prouse <rob@prouse.org", "Charlie Poole <charlie@poole.org", "NUnit" }));
```

Comment (JustinRChou) [inline src/NUnitFramework/tests/Attributes/AuthorTests.cs]: I am probably going to drop 
`Assert.That(authors, Has.Exactly(3).Items);`
as well.

And regarding TestCaseSource Author property, thoughts?

**Comment by CharliePoole** [comment]
Sorry, I missed the questions...

The XML can be tested by converting the test that is built to XML and looking at the elements. However, we test that properties are reported correctly in the XML elsewhere and you have tested here that the properties are correctly set, so you __should__ be OK. As a final manual test, look at the TestResul.xml for your run of the framework tests. You should see the multiple property entries for the fixture and the method that have multiple authors.

I'm not a fan of Author on TestCaseSource. I don't think there's a need to identify authorship at every level. Frankly, I think we already took it too far with TestCase.

File: src/NUnitFramework/tests/Attributes/AuthorTests.cs (hunk 1 above)
Comment (CharliePoole) [inline src/NUnitFramework/tests/Attributes/AuthorTests.cs]: Right. My change would drop the five lines. Or four if you use authors as a temporary.

**Comment by JustinRChou** [comment]
Made the changes.
I just found it odd that TestCase had more properties than TestCaseSource.

File: src/NUnitFramework/tests/Attributes/AuthorTests.cs (hunk 1 above)
Comment (CharliePoole) [inline src/NUnitFramework/tests/Attributes/AuthorTests.cs]: We have code that tests the count and then each item in some of our tests, but it's old code that pre-dates use of EqualConstraint with arrays.

**Comment by CharliePoole** [comment]
TestCase looks like it's similar to TestCaseSource but it represents __one__ test case, while TestCaseSource represents a bunch of them.

**Comment by CharliePoole** [comment]
Feel free to merge once CI passes.

**Pull request merged**